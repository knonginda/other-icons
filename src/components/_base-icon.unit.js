import BaseIcon from './_base-icon'

describe('@components/_base-icon', () => {
  it('renders its content', () => {
    const wrapper = shallowMount(BaseIcon, {
      propsData: {
        name: 'some-icon',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders its content when size is medium', () => {
    const wrapper = shallowMount(BaseIcon, {
      propsData: {
        name: 'some-icon',
        size: 'medium',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders its content when size is large', () => {
    const wrapper = shallowMount(BaseIcon, {
      propsData: {
        name: 'some-icon',
        size: 'large',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders its content when size is xsmall', () => {
    const wrapper = shallowMount(BaseIcon, {
      propsData: {
        name: 'some-icon',
        size: 'xsmall',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders its content when size is invalid', () => {
    const wrapper = shallowMount(BaseIcon, {
      propsData: {
        name: 'some-icon',
        size: 'invalid',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
