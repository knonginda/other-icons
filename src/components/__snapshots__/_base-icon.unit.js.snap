// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/_base-icon renders its content 1`] = `<span class="small baseIcon"><svg><use xlink:href="#some-icon"></use></svg></span>`;

exports[`@components/_base-icon renders its content when size is invalid 1`] = `<span class="small baseIcon"><svg><use xlink:href="#some-icon"></use></svg></span>`;

exports[`@components/_base-icon renders its content when size is large 1`] = `<span class="large baseIcon"><svg><use xlink:href="#some-icon"></use></svg></span>`;

exports[`@components/_base-icon renders its content when size is medium 1`] = `<span class="medium baseIcon"><svg><use xlink:href="#some-icon"></use></svg></span>`;

exports[`@components/_base-icon renders its content when size is xsmall 1`] = `<span class="xsmall baseIcon"><svg><use xlink:href="#some-icon"></use></svg></span>`;
